alert.go

```diff
@@ -39,32 +39,32 @@ func (a *alerters) appendAlerter(alerter Alerter) {
 }
 
 func (a *alerters) alertAll(ctx context.Context, botType BotType, err error) error {
-	alertErrs := &alertErrs{}
+	errs := &alertErrs{}
 	for _, alerter := range *a {
 		// Considering the irregular state of Bot's lifecycle and importance of alert,
 		// it is safer to be panic-proof.
 		func() {
 			defer func() {
 				if r := recover(); r != nil {
 					e, ok := r.(error)
-					var err error
 					if ok {
-						err = fmt.Errorf("panic on Alerter.Alert: %w", e)
-					} else {
-						err = fmt.Errorf("panic on Alerter.Alert: %+v", r)
+						errs.appendError(fmt.Errorf("panic on alerting via %T: %w", alerter, e))
+						return
 					}
-					alertErrs.appendError(err)
+
+					errs.appendError(fmt.Errorf("panic on alerting via %T: %+v", alerter, r))
 				}
 			}()
+
 			err := alerter.Alert(ctx, botType, err)
 			if err != nil {
-				alertErrs.appendError(err)
+				errs.appendError(fmt.Errorf("failed to send alert via %T: %w", alerter, err))
```

Comment:
While multiple `Alerter` implementations can be registered and be processed at once, the error returned by each called method itself does not tell which `Alerter` implementation caused the error. This, by wrapping the error with `Alerter`'s type information, enables the operator to distinguish which `Alerter` failed.